controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -334,14 +334,18 @@ public OperationContext createContext(String scope, String name) {
                 new AbstractMap.SimpleEntry<>(x.getKeyStart(), x.getKeyEnd())).collect(Collectors.toList());
 
         future.thenAccept(result -> {
-            DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_COUNT, scope, name),
-                    newSegments.size() - sealedSegments.size());
+            DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, name), newSegments.size());
             getSealedRanges(scope, name, sealedSegments, context, executor)
                     .thenAccept(sealedRanges -> {
-                        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_SPLITS, scope, name),
-                                findSplits(sealedRanges, newRanges));
-                        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_MERGES, scope, name),
-                                findSplits(newRanges, sealedRanges));
+                        long oldNumSplits = DYNAMIC_LOGGER.getCounterValue(nameFromStream(SEGMENTS_SPLITS, scope, name));
```

Comment:
another aspect to note here is -- getCounterValue from cache may be incorrect because 'manual' scale requests can be received on different controller instances.. they may each build their own cache and these counted values may diverge and we will have incorrect counted values on different controller instances.. 

so as a solution, we can always compute this value in context of this scale operation (as we have the metadata already read from the store) and not use a cache for these two counters. 
it will take care of both the problems of cache miss getting reset and computing incorrect values.